### PR TITLE
 add validation of kind name before saving in catalog

### PR DESCRIPTION
**Pull request by ShatalovYaroslav**

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalog/util/name/validator/BucketNameValidator.java [hunk 1]
```diff
@@ -23,33 +23,20 @@
...
 /**
- * @author ActiveEon Team
- * @since 11/13/2017
+ * According to this validator: the bucket name can be between 3 and 63 characters long, and can contain only lower-case characters, numbers, and dashes.
+ A bucket name must start with a lowercase letter and cannot terminate with a dash.
  */
 @Component
-public class BucketNameValidator {
+public class BucketNameValidator extends NameValidator {
     protected final static String VALID_BUCKET_NAME_PATTERN = "[a-z][a-z0-9-]{1,61}[a-z0-9]";
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/util/name/validator/BucketNameValidator.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: src/main/java/org/ow2/proactive/catalog/util/name/validator/KindNameValidator.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.stereotype.Component;
+
+
+/**
+ * According to this validator: the kind name can be between 1 and 61 characters long, and can contain only letters, numbers, underscores, dotes and dashes.
+ A kind names can be separated by slash symbol.
+ */
+@Component
+public class KindNameValidator extends NameValidator {
+    protected final static String VALID_KIND_NAME_PATTERN = "^([a-zA-Z0-9-_\\.]{1,61}\\/?)+$";
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/util/name/validator/KindNameValidator.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: src/main/java/org/ow2/proactive/catalog/util/name/validator/NameValidator.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 8/2/2018
+ */
+abstract public class NameValidator {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/util/name/validator/NameValidator.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/ow2/proactive/catalog/service/BucketService.java [hunk 4]
```diff
@@ -69,7 +69,7 @@ public BucketMetadata createBucket(String name) {
     }
 
     public BucketMetadata createBucket(String name, String owner) throws DataIntegrityViolationException {
-        if (!bucketNameValidator.checkBucketName(name)) {
+        if (!bucketNameValidator.checkName(name)) {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/service/BucketService.java]: I would prefer to call this method isValid instead of  checkName

**Pull request merged**